app/src/test/java/org/vss/impl/postgres/PostgresBackendImplIntegrationTest.java

```diff
@@ -0,0 +1,48 @@
+package org.vss.impl.postgres;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import org.jooq.DSLContext;
+import org.jooq.SQLDialect;
+import org.jooq.impl.DSL;
+import org.junit.jupiter.api.BeforeEach;
+import org.testcontainers.containers.PostgreSQLContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.vss.AbstractKVStoreIntegrationTest;
+
+@Testcontainers
+public class PostgresBackendImplIntegrationTest extends AbstractKVStoreIntegrationTest {
+
+  private final String POSTGRES_TEST_CONTAINER_DOCKER_IMAGE = "postgres:15";
+
+  @Container
+  private final PostgreSQLContainer postgreSQLContainer =
+      new PostgreSQLContainer(POSTGRES_TEST_CONTAINER_DOCKER_IMAGE)
+          .withDatabaseName("postgres")
+          .withUsername("postgres")
+          .withPassword("postgres");
+
+  @BeforeEach
+  public void initEach() throws Exception {
+
+    // This is required to get postgres driver in classpath before we attempt to fetch a connection
+    Class.forName("org.postgresql.Driver");
+    Connection conn = DriverManager.getConnection(postgreSQLContainer.getJdbcUrl(),
+        postgreSQLContainer.getUsername(), postgreSQLContainer.getPassword());
+    DSLContext dslContext = DSL.using(conn, SQLDialect.POSTGRES);
+
+    this.kvStore = new PostgresBackendImpl(dslContext);
```

Comment:
don't we need to close the previous DB connection, otherwise we may run out of file descriptors or such if we have many tests?

Reply:
Each and every test spins up new db cluster/instance so db connection shouldn't really be a problem. 
But I added a AfterEach block to destroy connection in any case.

For production, we are using connection pool and DSL.using(dataSource, dialect) where jooq/pool does the connection management for us.